### PR TITLE
Adds breaking change and best practice docs

### PR DESCRIPTION
**Pull request by thatguynamedandy**
# Description

I've added some additional documentation around breaking changes, and also some of the ideologies that we have adopted while building the library. I'm not sold on the word 'ideology' so open to ideas on alternatives. I mostly wanted to get across that Canopy is opinionated in it's approach and where possible it should be used as-is and not modified. This opinionated mindset hopefully gives strengths in terms of speed and consistency but looses slightly in terms of flexibility.

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/CONTRIBUTING.md#conventional-commits)
- [ ] I have provided an adequate amount of test coverage
- [ ] I have added the functionality to the [test app](./blob/master/CONTRIBUTING.md#build-test-application)
- [ ] I have provided a story in storybook to document the changes
- [ ] I have provided documentation in the notes section of the story
- [ ] I have added any new public feature modules to public-api.ts
- [ ] I have [linked the new component](<(https://github.com/Legal-and-General/canopy/blob/master/CONTRIBUTING.md#invision-dsm)>) to adobe DSM (if appropriate)

### CODE REVIEW

File: docs/BREAKING_CHANGES.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+A good example of this is the date component, in it's simplest form the component can be used like below.
+
+```html
+<lg-date-field formControlName="date">
+  Date of birth
+<lg-date-field>
+```
+
+The markup rendered to the browser is of course much more complex and consists of multiple input fields and other DOM elements. Theoretically we could change the encapsulated markup without causing a breaking change as the component api remains unchanged.
```
Comment (owensgit) [inline docs/BREAKING_CHANGES.md:67]: Minor, but...

```suggestion
The markup rendered to the browser is of course much more complex and consists of multiple input fields and other DOM elements. Theoretically we could change the encapsulated markup without causing a breaking change as the component API remains unchanged.
```

File: docs/BREAKING_CHANGES.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+This does however mean that the DOM and associated CSS classes can not be considered stable.
+
+We recommend that you do not override css styles in your application, if you do this you are likely to be impacted by changes which were not considered to be a breaking change.
```
Comment (owensgit) [inline docs/BREAKING_CHANGES.md:71]: I wonder if we should make the first part of this bold, to highlight the importance? 

```suggestion
We recommend that you **do not override css styles in your application**, if you do this you are likely to be impacted by changes which were not considered to be a breaking change.
```

File: docs/IDEOLOGIES.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Ideologies
+
+- Canopy is an opinionated design system, it is intended to be a set of rigid tested design patterns. 
+
+- In most cases it should not be extended or modified within applications.
+
+- There should be a 1 to 1 mapping between the design and development implementation.
```
Comment (owensgit) [inline docs/IDEOLOGIES.md:7]: Wonder if the we could link to the DSM in here?

```suggestion
- There should be a 1 to 1 mapping between the [design](https://legalandgeneral.invisionapp.com/dsm/legalandgeneral/canopy?mode=preview) and development implementation.
```

File: docs/IDEOLOGIES.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Ideologies
```
Comment (owensgit) [inline docs/IDEOLOGIES.md:1]: Thinking about, I wonder if **Best Practice**  is better name?

**Review by owensgit — COMMENTED** [review]
Looks great, just a few suggestions from my side.

File: docs/IDEOLOGIES.md (hunk 4 above)
Comment (thatguynamedandy) [reply docs/IDEOLOGIES.md]: Yep much better 👍  thanks!

**Review by elenagarrone — APPROVED** [review]

**Review by owensgit — APPROVED** [review]

**Pull request merged**